core/remote/watcher.js

```diff
@@ -334,6 +334,13 @@ class RemoteWatcher {
             remoteChange.includeDescendant(change, previousChange)
             continue
           }
+        } else if (previousChange.type === 'DirTrashing' && previousChange.was._id === change.doc._id) {
```

Comment:
We're doing 2 different things:
1. on line 243, we make sure we don't trash a document that is at the destination path of another action (like a move)
2. on line 337, we make sure we don't trash a document that is the source of another change (like a move)

Reply:
Then maybe it would be less confusing to match on path, not _id, so `DirTrashing(foo)` cannot end up being squashed with `DirMove(FOO => BAR)` into `DirMove(foo => BAR)` on case-insensitive platforms. Although I suppose it should not happen anyway since we couldn't have both `foo` and `FOO` in Pouch...

Reply:
I see what you mean. I'll try to come up with a test for that.